File: apps/synapse-api/synapse_api/domain/entities/purchases.py
from dataclasses import dataclass, field
from datetime import date
from decimal import Decimal

from .common import Currency, Money, OrderStatus, Region


@dataclass(frozen=True)
class PurchaseItem:
    material_name: str
    quantity: int
    unit_cost: Money
    total: Money


@dataclass(frozen=True)
class PurchaseOrder:
    id: str
    vendor_name: str
    region: Region
    status: OrderStatus
    date: date
    items: list[PurchaseItem] = field(default_factory=list)
    total: Money = field(
        default_factory=lambda: Money(amount=Decimal(0), currency=Currency("INR"))
    )


@dataclass(frozen=True)
class VendorSummary:
    vendor_name: str
    total_orders: int
    total_spend: Money
    on_time_rate: float

File: apps/synapse-web/src/domain/entities/purchases.ts
import type { OrderStatus, Region } from "./common";

export interface PurchaseItem {
	material_name: string;
	quantity: number;
	unit_cost: number;
	currency: string;
	total: number;
}

export interface PurchaseOrder {
	id: string;
	vendor_name: string;
	region: Region;
	status: OrderStatus;
	date: string;
	items: PurchaseItem[];
	total: number;
	currency: string;
}

export interface VendorSummary {
	vendor_name: string;
	total_orders: number;
	total_spend: number;
	currency: string;
	on_time_rate: number;
}

export interface PurchaseOrderListResponse {
	items: PurchaseOrder[];
	total: number;
	page: number;
}
